R/available.R

```diff
@@ -114,3 +109,39 @@ suggest <- function(path = ".",  field = c("Title", "Description"), text = NULL)
 
   namr(text)
 }
+
+
+
+check_online_terms <- function(term, urban = TRUE) {
+  if (urban) {
+    compact(list(
+      get_bad_words(term),
+      get_abbreviation(term),
+      get_wikipidia(term),
+      get_wiktionary(term),
+      get_urban_data(term),
+      get_sentiment(term)))
+  } else {
+    compact(list(
+      get_bad_words(term),
+      get_abbreviation(term),
+      get_wikipidia(term),
+      get_wiktionary(term),
+      get_sentiment(term)))
+  }
+}
+
+
+check_urban <- function() {
+  if (interactive()) {
+    cat("Urban Dictionary can contain potentially offensive results,\n  should they be included? [Y]es / [N]o:\n")
+    result <- tryCatch(scan("", what = "character", quiet = TRUE, nlines = 1), error = function(x) "N")
+    if (!identical(toupper(result), "Y")) {
```

Comment:
I think it would be clearer to use an early return here, e.g.

```r
check_urban <- function() {
  if (!interactive()) {
     return(TRUE)
  }
  
  cat("Urban Dictionary can contain potentially offensive results,\n  should they be included? [Y]es / [N]o:\n")
  result <- tryCatch(scan("", what = "character", quiet = TRUE, nlines = 1), error = function(x) "N")
  identical(toupper(result), "Y"))
}
```